### ISSUE TITLE
Deconvolution Layer ifm != ofm

### ISSUE THREAD

**Issue by wr0112358** [ARM-software/ComputeLibrary#453]
**Output of 'strings libarm_compute.so | grep arm_compute_version':**
arm_compute_version=v18.05 Build options: {'arch': 'arm64-v8a', 'opencl': '1', 'neon': '1', 'asserts': '1', 'debug': '1', 'os': 'android', 'Werror': '1'} Git hash=b3a371bc429d2ba45e56baaf239d8200c2662a74

**Platform:**
hikey960

**Operating System:**
android

**Problem description:**
ACL Deconvolution in both NEON and CL versions fails for the case ifm != ofm.

Configuration tensorflow:
tf.nn.conv2d_transpose
ifm=4
ofm=2
w/h=8
stride=2
ksize=4
padding=VALID

input pattern: [ 0...0] [1...1] [2...2] [3...3]
filters: bilinear upsampling

Configuration ACL:
same except padding=3

Output tensorflow:
result: (1, 2, 18, 18),
[[[[0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]
  [0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.  ]]

 [[0.0625 0.1875 0.25  0.25  0.25  0.25  0.25  0.25  0.25 0.25  0.25  0.25  0.25  0.25  0.25  0.25  0.1875 0.0625]
  [0.1875 0.5625 0.75  0.75  0.75  0.75  0.75  0.75  0.75 0.75  0.75  0.75  0.75  0.75  0.75  0.75  0.5625 0.1875]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.25  0.75  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0.75  0.25 ]
  [0.1875 0.5625 0.75  0.75  0.75  0.75  0.75  0.75  0.75 0.75  0.75  0.75  0.75  0.75  0.75  0.75  0.5625 0.1875]
  [0.0625 0.1875 0.25  0.25  0.25  0.25  0.25  0.25  0.25 0.25  0.25  0.25  0.25  0.25  0.25  0.25 0.1875 0.0625]]]]

Output ACL:
shape=(18, 18, 2)
[ 0.1875 0.5625 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.5625 0.1875 ]
[ 0.5625 1.6875 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 1.6875 0.5625 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.75 2.25 3 3 3 3 3 3 3 3 3 3 3 3 3 3 2.25 0.75 ]
[ 0.5625 1.6875 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 2.25 1.6875 0.5625 ]
[ 0.1875 0.5625 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.75 0.5625 0.1875 ]

[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ]

**Comment by wr0112358** [comment]
After calculating the convolution by hand and coming to the same results, I checked the documentation, which gives the solution: "The 4d weights with dimensions [width, height, IFM, OFM]"

Sorry for the noise.
